### ISSUE TITLE
add buttons to layers panel to rearrange layers

### ISSUE THREAD

**Issue by qgib** [qgis/QGIS#22080]
Author Name: **Jesse McMillan** (Jesse McMillan)
Original Redmine Issue: [14070](https://issues.qgis.org/issues/14070)

Redmine category:gui

---
Drag-and-drop is sometimes not a very good way of rearranging layers. I frequently end up adding a layer to a group when I mean to put it above the group. I think having a "move layer up" and "move layer down" set of buttons on the layers panel would be helpful.

**Comment by wiesehahn** [comment]
This feature was requested more than 7 years ago :)
I think it would still add functionality. 

Currently there is a limited number of keyboard shortcuts implemented for the Layer Panel.
for example we have: 

<kbd>Cmd + D</kbd> to delete a layer

It would be nice to have additional keyboard shortcuts implemented here. As <kbd>Cmd + D</kbd> has the same functionality for a layer as it has e.g. in VSCode for a code line, it might make sense to add further shortcuts on this basis. 

<kbd>Alt + Shift +Up/Down</kbd> to doublicate a layer above/below
<kbd>Alt +Up/Down</kbd> to move a layer or group up/down
...

**Comment by andreasneumann** [comment]
> This feature was requested more than 7 years ago :)
> I think it would still add functionality.

Well yes - there are many such requests. We have literally thousands of such feature requests.

Just to add a little background:
QGIS has no budget to work on feature requests. The budget we have available is spent on infrastructure, bug fixing, documentation, code refactoring, contributor meetings, etc.

If you would like a feature request to be implemented you have some options:
- implement it yourself and create a pull request
- ask a friend to do so
- ask a commercial support provider to do it as a paid contract (see https://www.qgis.org/en/site/forusers/commercial_support.html)
- wait until someone implements the feature request by chance (highly unlikely, unless it is really annoying someone with the skills or funds to implement it)

**Comment by wiesehahn** [comment]
Yes, I was wondering if Github Issues should be used for feature requests in QGIS (or just bugs. But as there is the label and many requests are written here, I thought it may be used to kind of filter/sort requests by the community. I don't expect it to be implemented, was just meant as a hint possibly for future UI improvements :)

**Comment by andreasneumann** [comment]
There is nothing wrong with opening feature requests. I don't want to discourage that. It helps to discuss ideas and bring together people who can potentially implement or fund it together.

However, I just wanted to mention the fact, that just opening the feature request doesn't mean it will be automatically implemented for the above listed reasons.

It often comes down to someone taking the initiative. QGIS.ORG can't do it (not enough resources). But individual users or professional users (or QGIS user groups) can pick the feature request up and fund or implement it.

**Comment by gacarrillor** [comment]
Duplicate of https://github.com/qgis/QGIS/issues/50829
(Please reopen if you don't consider it a duplicate)
